src/Interpreters/MySQL/InterpretersMySQLDDLQuery.cpp

```diff
@@ -92,22 +92,40 @@ static NamesAndTypesList getColumnsList(const ASTExpressionList * columns_defini
         }
 
         ASTPtr data_type = declare_column->data_type;
+        auto * data_type_function = data_type->as<ASTFunction>();
 
-        if (is_unsigned)
+        if (data_type_function)
         {
-            auto * data_type_function = data_type->as<ASTFunction>();
+            String type_name_upper = Poco::toUpper(data_type_function->name);
 
-            if (data_type_function)
+            if (is_unsigned)
             {
-                String type_name_upper = Poco::toUpper(data_type_function->name);
-
                 /// For example(in MySQL): CREATE TABLE test(column_name INT NOT NULL ... UNSIGNED)
-                if (type_name_upper.find("INT") != std::string::npos && !endsWith(type_name_upper, "SIGNED")
+                if (type_name_upper.find("INT") != String::npos && !endsWith(type_name_upper, "SIGNED")
                     && !endsWith(type_name_upper, "UNSIGNED"))
                     data_type_function->name = type_name_upper + " UNSIGNED";
             }
-        }
 
+            /// Transforms MySQL ENUM's list of strings to ClickHouse string-integer pairs
+            /// For example ENUM('a', 'b', 'c') -> ENUM('a'=1, 'b'=2, 'c'=3)
+            /// Elements on a position further than 32767 are assigned negative values, starting with -32768.
+            /// Note: Enum would be transformed to Enum8 if number of ellements is less then 128, otherwise it would be transformed to Enum16.
```

Comment:
```suggestion
            /// Note: Enum would be transformed to Enum8 if number of elements is less then 128, otherwise it would be transformed to Enum16.
```